blockchain/fullblocktests/generate.go

```diff
@@ -543,6 +544,17 @@ func Generate(includeLargeReorg bool) (tests [][]TestInstance, err error) {
 	tests = append(tests, testInstances)
 	g.AssertTipHeight(uint32(coinbaseMaturity) + 1)
 
+	// TODO: Test is not failing as expected
+	// Create block with a ticket purchase before stake enabled height.
+	//
+	//   genesis -> bp
+	// g.NextBlock("bes1", nil, nil, func(b *wire.MsgBlock) {
+	// 	spendOut := chaingen.MakeSpendableOut(g.Tip(), 0, 2)
+	// 	ticketPrice := dcrutil.Amount(g.CalcNextRequiredStakeDifficulty())
+	// 	g.CreateTicketPurchaseTx(&spendOut, ticketPrice, lowFee)
+	// })
+	// rejected(blockchain.ErrInvalidEarlyStakeTx)
```

Comment:
I should also note that this is not testing the same thing as the one it's replacing.  The test it's replacing tries to cast a vote too early.  If you look below slightly, there are TODO comments, one of which is `// - Try to vote before stake validation height`.  It should be there.